supplier_app/tests/test_view.py

```diff
@@ -913,54 +939,61 @@ def test_GET_edit_address_view(self):
         )
 
     def test_post_edit_address_info_as_ap(self):
-        request = self.factory.post(
+        response = self.client.post(
             reverse(
                 self.address_edit_url,
-                kwargs=self.kwargs_address
+                kwargs=self.kwargs
             ),
             data=self.ADDRESS_POST
         )
-        request.user = self.ap_user
-        response = EditAddressView.as_view()(request, **self.kwargs_address)
         self.assertEqual(HTTPStatus.FOUND, response.status_code)
         self.assertEqual(
             reverse(
                 self.taxpayer_detail_url,
-                kwargs=self.kwargs_taxpayer
+                kwargs=self.kwargs_taxpayer_id,
             ),
             response.url
         )
 
-    def test_post_edit_address_info_as_supplier_redirects_to_supplier_home(self):
-        request = self.factory.post(
+    def test_post_edit_address_info_as_supplier_redirects_to_supplier_details(self):
+        response = self.client2.post(
```

Comment:
Remember what we talked on our Code Review Session about having all specific resources tied to the unit test.

Reply:
done